### ISSUE TITLE
[Fetaure Request] Layer normalization for NCHW/NHWC with fast gpu kernel

### ISSUE THREAD

**Issue by x10000year** [tensorflow/tensorflow#14523]
tf.contrib.layers.layer_norm is slow and only supports NHWC layout. It is beneficial to have a fast gpu kernel for layer normalization that supports both NCHW and NHWC. I think layer normalization is quite useful when the minibatch is extremely small or comprises only one sample (in which case batch norm/renorm doesn't work) and when local receptive fields is desired (in which case instance norm is bad).

**Comment by MarkSonn** [comment]
Why did you close this issue immediately? I'm wondering the same, what did you discover?
